include/exadg/convection_diffusion/driver.cpp

```diff
@@ -238,7 +237,23 @@ Driver<dim, Number>::solve()
 
         // Adapt the mesh before post_solve(), in order to recalculate the
         // time step size based on the new mesh.
-        do_adaptive_refinement(time_integrator->get_number_of_time_steps());
+        if(trigger_coarsening_and_refinement_now(
+             application->get_parameters().amr_data.trigger_every_n_time_steps,
+             time_integrator->get_number_of_time_steps()))
+        {
+          // AMR is only implemented for implicit timestepping.
+          std::shared_ptr<TimeIntBDF<dim, Number>> bdf_time_integrator =
+            std::dynamic_pointer_cast<TimeIntBDF<dim, Number>>(time_integrator);
+
+          AssertThrow(bdf_time_integrator.get(),
+                      dealii::ExcMessage("Combination of adaptive mesh refinement "
+                                         "and explicit time integration not implemented."));
+
+          mark_cells_coarsening_and_refinement(*grid->triangulation,
```

Comment:
if we were to introduce `get_solution_np` in the time integrator base class, we could get rid of the dynamic casts. For explicit timestepping, I would have to introduce another one in the future.

Reply:
I think this is ok for now. However, one could change the exc-message and write that "only BDF time integration is implemented for AMR", or similar. (I think the present message is only valid as long as no other than the two time stepping routines currently available for `ConvDiff` is implemented.)

Reply:
coming with the next push tomorrow morning :+1: